modules/ROOT/pages/genai-integrations.adoc

```diff
@@ -0,0 +1,291 @@
+Therefore, it is recommended to see the respective provider's documentation for details on, for example, the maximum number of embeddings that can be generated per request.
+====
+
+.Signature for `genai.vector.encodeBatch()` label:procedure[]
+[source,syntax,role="noheader",indent=0]
+----
+genai.vector.encodeBatch(resources :: LIST<STRING>, provider :: STRING, configuration :: MAP = {}) :: (index :: INTEGER, resource :: STRING, vector :: LIST<FLOAT>)
+----
+
+* The `resources` (a `LIST<STRING>`) parameter is the list of objects to transform into embeddings, such as chunks of natural language text.
+* The `provider` (a `STRING`) is the case-insensitive identifier of the provider to use.
+See <<ai-providers>> for supported options.
+* The `configuration` (a `MAP`) specifies provider-specific settings such as which model to invoke, as well as any required API credentials.
+See <<ai-providers>> for details of each supported provider.
++
+[NOTE]
+====
+Because this argument may contain sensitive data, it is obfuscated in the _query.log_.
+However, if the function call is misspelled or the query is otherwise malformed, it may be logged without being obfuscated.
+====
++
+Each returned row contains the following columns:
+
+* The `index` (an `INTEGER`) is the index of the corresponding element in the input list, to aid in correlating results back to inputs.
+* The `resource` (a `STRING`) is the name of the input resource.
+* The `vector` (a `LIST<FLOAT>`) is the generated vector embedding for this resource.
+
+.Generate embeddings for multiple chunks of a larger text
+====
+
+Given a list of page texts, you can generate an embedding for each of the pages with a single procedure call and create a corresponding graph structure:
+
+.Query
+[source,cypher]
+----
+CREATE (b:Book { title: $bookTitle })
+WITH book
+CALL genai.vector.encodeBatch($pageTexts, "VertexAI", { token: $token, projectId: $project }) YIELD index, resource, vector
+WITH book, index, resource, vector
+CREATE (:Page { index: index, text: resource, vector: vector })-[:OF]->(book)
+----
+====
+
+.Generate embeddings for many text properties
+====
+
+If you want to generate embeddings for the text content of all nodes with label `Tweet`, you can use `CALL ... IN TRANSACTIONS` to split the work up into batches and issue one API request per batch.
+
+Assuming nodes with the `Tweet` have a `text` property, you can generate vector embeddings for each one and write them to the `embedding` property on each one in batches of 1000, for example:
+
+.Query
+[source,cypher]
+----
+MATCH (n:Tweet)
+WHERE n.text IS NOT NULL
+CALL {
+    WITH n
+    WITH collect(n) AS nodes, collect(n.text) AS resources
+    CALL genai.vector.encodeBatch(resources, "VertexAI", { token: $token, projectId: $project }) YIELD index, vector
+    CALL db.create.setNodeVectorProperty(nodes[index], "vector", vector)
+} IN TRANSACTIONS OF 1000 ROWS
+----
+====
+
+[[ai-providers]]
+== AI Providers
```

Comment:
I think I personally would prefer it as "GenAI" / "Generative AI" and not just "AI" - the latter can be somewhat more controversial.